### DISCUSSION TITLE
Can I create filtered percentage stacked bars using Plot?

### DISCUSSION THREAD

**Discussion by mkfreeman** [observablehq/plot#2198] [Q&A] [answered]
Is it possible to create a plot that uses `offet: 'normalize'` to display percentages AND THEN ALSO filters down the data so you only see the desired percentages (with the y axis reset to the remaining values)? The intended output would be a chart like this where the y-axis only extends to the range of the displayed data:
<img width="636" alt="Screenshot 2024-10-14 at 10 56 02 AM" src="https://github.com/user-attachments/assets/a7911822-c732-4c00-9e1f-db6904b38a45">

I've tried a few approaches in [this notebook](https://observablehq.com/d/761b3c6d859d71bc), the closest I got was to filter after stacking, but that doesn't recompute the `y1` and `y2` values of remaining categories:

```js
Plot.plot({
  marks: [
    Plot.barY(tidy, {
      filter: (d) => d.age !== "<10" && d.age !== "10-19",
      ...Plot.stackY({
        offset: "normalize",
        x: "state",
        y: "population",
        fill: "age",
        sort: { color: null, x: "-y" }
      })
    })
  ]
})

```
<img width="639" alt="Screenshot 2024-10-14 at 10 58 15 AM" src="https://github.com/user-attachments/assets/9ff7ea9d-a246-4eb9-a4f3-7d11351b942f">

I understand this is rather niche so the answer may be in the data transformation I performed outside of Plot (totally reasonable!)

**Comment by Fil** [comment, marked as answer]
You need a [custom offset function](https://observablehq.com/plot/transforms/stack#stack-options) that takes as basis the maximum value from the whole series and filters out the meanies:

```js
Plot.plot({
  marks: [
    Plot.barY(tidy, {
      offset: function (facets, Y1, Y2, Z) {
        for (const series of facets) {
          for (const I of series) {
            const basis = d3.max(I, (i) => Y2[i]);
            for (const i of I)
              if (Z[i] === "<10" || Z[i] === "10-19") {
                Y1[i] = Y2[i] = NaN;
              } else {
                Y1[i] /= basis;
                Y2[i] /= basis;
              }
          }
        }
      },
      x: "state",
      y: "population",
      fill: "age",
      order: (d) => d.age === "<10" || d.age === "10-19", // you can dispense with this if you work more on the offset function :)
      sort: { color: null, x: "-y" }
    })
  ]
})
```

**Reply by mkfreeman** [reply]
Thanks so much @Fil , I hadn't seen that before!
